department-of-veterans-affairs/vets-api
# feat: Use Sidekiq Utilization for HPA

https://github.com/department-of-veterans-affairs/va.gov-team/issues/89549

This provides Sidekiq a utilization metric to scale on rather than scaling once utilization is already at 100% and a line is forming. The goal for this value will be to scale around 60-80%.

## Testing

1. Checkout this branch locally
2. `cd` to `apps/vets-api/prod` in the manifests repo
3. Update your Chart.yaml repository to point to your local vets-api (e.g. `repository: "file://../../../../vets-api/chart"`)
4. Update your chart version to 0.13.3 (in both places in Chart.yaml) and then run `helm dependency update`
5. `curl` DataDog for the last 10 mins using the new HPA query:
```
HPA_QUERY=$(helm template . | grep -A 9 sidekiq-utilization | grep query: | cut -f 4- -d' ')
CURRENT_TIME=$(date +%s)

curl -s --max-time 2 -G \
  'https://vagov.ddog-gov.com/api/v1/query' \
  --data-urlencode "query=$HPA_QUERY" \
  -d "from=$((CURRENT_TIME - 600))" \
  -d "to=${CURRENT_TIME}" \
  -H "DD-API-KEY: $DD_API_KEY" \
  -H "DD-APPLICATION-KEY: $DD_APP_KEY" | jq
```

Values in the point list (epoch, metric) should match [this chart](https://vagov.ddog-gov.com/s/f327ad72-c02a-11ec-a50a-da7ad0900007/zub-r2c-zmz).
